Готова 5 проектная работа "Карты подскажут"

SmithDaria opened the pull request.

SmithDaria committed:
Готова 5 проектная работа "Карты подскажут"

Avaakum commented on index.html:
```diff
@@ -1,22 +1,176 @@
+        <p class="content__paragraph">
+          И кстати, сейчас необязательно охотиться за редкими физическими
+          колодами Oblique Strategies — тем более, что их моментально разбирают
+          фанаты Ино. Есть довольно симпатичные веб-сайты с
```
Слово "веб-сайты" так же должно быть ссылкой из макета

Avaakum commented on index.html:
```diff
@@ -1,22 +1,176 @@
+            href="https://apps.apple.com/us/app/oblique-strategies-se/id1104927011"
+            target="_blank"
+          >
+            iOS
```
Между тегов ссылки должно быть "для iOS"

Avaakum commented on styles/style.css:
```diff
@@ -19,4 +17,228 @@ ol,
+  padding-top: 10%;
+}
+.heading {
+  font-family: var(--accent-font, sans-serif);
```
Это шрифт с засечками, для второго шрифта  "STIX Two Text" нужен как фолбэк тип с засечками, serif

Avaakum commented on index.html:
```diff
@@ -1,22 +1,176 @@
+    <header class="header">
+      <h1 class="heading content-section header__title">Карты подскажут</h1>
+      <nav class="header__menu content-section">
+        <div class="header__logo">
```
---

**Можно лучше:**

Логотип должен быть ссылкой **без атрибута href,**(что бы по ссылке не было перехода). А так же, что бы при навигации с клавиатуры этот элемент не получал видимый фокус. **Пустой атрибут href тоже нужно удалить.**

Avaakum commented on index.html:
```diff
@@ -1,22 +1,176 @@
+        </p>
+      </div>
+      <figure class="content__figure">
+        <picture class="content__picture">
```
---

**Отлично:**

Отличная семантика

Avaakum commented on index.html:
```diff
@@ -1,22 +1,176 @@
+            srcset="./images/cards-1x.webp 1x, ./images/cards-2x.webp 2x"
+            type="image/webp"
+          >
+          <img
```
Тут так же стоит использовать srcset и указать 2 изображения в значении атрибута. Вместо первого source и img, тогда будет только img c srcset  вместо src. На ошибку валидатора в данном случае можно не обращать внимание

Avaakum commented on index.html:
```diff
@@ -1,22 +1,176 @@
+          >
+          <source
+            srcset="./images/cards-1x.avif 1x, ./images/cards-2x.avif 2x"
+            type="image/avif"
```
порядок форматов должен быть от самого прогрессивного и оптимизированного до самого старого формата. У тега img должны быть изображения png.

Avaakum commented on index.html:
```diff
@@ -1,22 +1,176 @@
+        <p class="content__paragraph">
+          Вот что сам Брайан Ино рассказывает о создании Oblique Strategies:
+        </p>
+        <blockquote class="content__blockquote">
```
---

**Отлично:**

Отличная семантика

Avaakum commented on styles/style.css:
```diff
@@ -19,4 +17,228 @@ ol,
+a:focus-visible {
+  outline: 1px solid currentColor;
+}
+.page {
```
Дубликатов обращений к селекторам не следует допускать в проекте, данный подход увеличивает вес правил написанных в селекторе ниже по коду, правила написанные в селекторе выше могут перебиваться правилами описанными в селекторе ниже. Такой подход усложняет разработку и поддержку кода. 
Пожалуйста, проконтролируйте по всему файлу, что бы обращение к селектору использовалось в проекте единожды. Для удобства навигации можно использовать поиск по строке.

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Доброго времени суток. У вас получилась хорошая работа!

_Что понравилось:_

- работают темы
- отличная структура файлов проекта
- Хорошая семантика
- Хорошие стили 
- Отзывчивая верстка
- Похоже на макет на брейк-поинтах


_Что необходимо исправить:_

- **Все комментарии, кроме комментариев с пометками «Можно лучше» и «Отлично». А так же не забудьте проверить, есть ли свёрнутые комментарии, которые обычно находятся по центру страницы с ревью при прокрутке вниз, и выглядят следующим образом https://disk.yandex.ru/i/FyX-8tzeqLjimw**



- При ширине экрана 1440px _не совпадает с макетом_:
   - https://disk.yandex.ru/i/rEhIqAVQ2gwwJg выравнивание кнопки

- При ширине экрана 360px _не совпадает с макетом_:
  - https://disk.yandex.ru/i/lKZZqcxEhBAScQ выравнивание 

- https://disk.yandex.ru/i/B4cfnQ9bedRdkQ  Маркеры списка должны находится по центру  напротив элемента списка, а не по верхней строке. Для этого нужно опустить вниз буллит на  50% элемента списка, например с помощью  inset-block-start и затем поднять его на половину собственной высоты(7.5px) с помощью margin-block-start. 


_Что можно улучшить:_

- Все комментарии с пометкой "Можно лучше"
- Для всех функций ` var() ` в CSS нужно задать резервные значения,  на случай если доступ к кастомному свойству по имени будет невозможен. 
- больше использовать currentColor, где это возможно
- больше использовать vi вместо vw


Желаю успеха!

SmithDaria committed:
Отправляю 5 проект "Карты подскажут" после ревью

SmithDaria commented:
Отправляю на повторное ревью

Avaakum commented:
Развернуть общий комментарий ⬇︎

Здравствуйте. К сожалению проект отклоняется от проверки. Итерация не теряется. 





- _На повторных итерациях не исправлены критические замечания._ 
    - **Пример:** 
         - Начиная с неисправленных пунктов в общем комментарии https://disk.yandex.ru/i/2v6K71vSnG4B5A  в ревью предыдущей итерации нужно проверить исправление **всех критических комментариев** в работе. Если у вас есть сомнения по поводу того, что комментарий исправлен, то, пожалуйста, перечитайте комментарии внимательно, там описано, что нужно исправить в работе. Если что-то исправить не получается, то можно смело обращаться к наставнику, он сможет подсказать. Чтобы исправить ошибки валидатора(при наличии), можно в chrome перевести страницу на русский, чтобы было понятнее, что именно за ошибки он выдаёт. А для соответствия макету можно воспользоваться [расширением PixelPerfect](https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi  )  для Chrome.



     - Пожалуйста, будьте внимательны, проверяйте, исправляйте и присылайте работу только после исправления **всех критических замечаний** с  предыдущей итерации
    - Если у вас есть технические проблемы, то попробуйте обратиться в поддержку или спросить в чате студентов не было ли у кого похожих случаев.

SmithDaria committed:
Отправляю на ревью 5 проект "Карты подскажут"

Прошу разрешить еще одно ревью, так как в прошлый раз прикрепила старый файл стилей и вообще не туда. Это больше относится к незнанию гитхаба, прошу войти в положение. По пиксель перфект у меня все совпадает.

SmithDaria commented:
Прошу разрешить еще одно ревью, так как в прошлый раз прикрепила файл стилей вообще не туда. Это больше относится к незнанию гитхаба, прошу войти в положение. По пиксель перфект у меня все совпадает.

Avaakum commented on style.css:
```diff
@@ -0,0 +1,249 @@
+*,
```
Лишний файл

Avaakum reviewed (approved):
Развернуть общий комментарий ⬇︎

Здравствуйте! Рад вам сообщить, что ваша работа принята. Сейчас страничка выглядит прекрасно, всё работает отлично. Отличная семантика и структура. Желаю вам удачи в следующих спринтах и проектах! 



---


## Перед началом следующего спринта, пожалуйста, поправьте критический момент в коде


---




Ссылки:

- Можно использовать неразрывные пробелы для того чтобы текст не переносился в неожиданных местах. Неразрывные пробелы заменяют обычные, если нужно чтобы какая-то фраза не переносилась, например не нужно чтобы "на собаках" раскидывалось по разным строчкам при некоторых ширинах экранов, тогда `«на&nbsp;собаках»`. Но если страница с адаптацией, то всегда стоит проверять как текст с пробелами выглядит на всех ширинах.
- Про альтернативный текст для картинок https://habr.com/ru/post/341810/
- [Ссылка про подгрузку шрифтов](https://only-to-top.ru/blog/tools/2019-03-13-nastrojte-pokaz-vsego-teksta-vo-vremya-zagruzki-veb-shriftov.html) 



Несколько  ссылок на материал по теме нативных CSS переменных от авторов практикума:

- [1 часть](https://tproger.ru/articles/custom-properties-in-css-part-1/)

- [2 часть](https://tproger.ru/articles/custom-properties-in-css-part-2/)

- [3 часть](https://tproger.ru/articles/custom-properties-in-css-part-3/)
